src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -118,19 +116,23 @@ public boolean Disconnect(String peerId){
     public boolean  DisconnectIncomingConnections(){
 
         boolean ret = false;
+
         if (mServerSocketList != null) {
-            for(int i = 0; i < mServerSocketList.size(); i++) {
-                BtToServerSocket tmp = mServerSocketList.get(i);
-                if (tmp != null) {
-                    print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
-                    tmp.Stop();
-                    tmp = null;
-                    mServerSocketList.set(i, tmp);
-                    ret = true;
+            synchronized(mServerSocketList) {
+
+                for (int i = 0; i < mServerSocketList.size(); i++) {
+                    BtToServerSocket tmp = mServerSocketList.get(i);
+                    if (tmp != null) {
+                        print_debug("Disconnect:::Stop : mBtToServerSocket :" + tmp.getName());
+                        tmp.Stop();
+                        tmp = null;
+                        mServerSocketList.set(i, tmp);
+                        ret = true;
```

Comment:
:+1: 

I'm going to assume this will get fixed in 0.8 since the current functionality won't work there. The return value is only used for kill connection and in 0.8 kill connection needs to be peer specific. So you will have to change this logic there.